### PR TITLE
telegraf: add a more useful test

### PR DESCRIPTION
**Pull request by sparrc**
follow-up to PR #43076 to add more useful tests

### CODE REVIEW

File: Library/Formula/telegraf.rb [hunk 1]
```diff
@@ -77,5 +77,8 @@ def plist; <<-EOS.undent
 
   test do
     assert_match /Telegraf/, shell_output("telegraf -version")
+    system("telegraf -sample-config > #{testpath}/config.toml")
```
Comment (mikemcquaid) [inline Library/Formula/telegraf.rb]: Please use `shell_output` and `(testpath/"config.toml").write` to write out the file, thanks!

File: Library/Formula/telegraf.rb [hunk 2]
```diff
@@ -77,5 +77,8 @@ def plist; <<-EOS.undent
 
   test do
     assert_match /Telegraf/, shell_output("telegraf -version")
```
Comment (mikemcquaid) [inline Library/Formula/telegraf.rb]: You can kill this test now.

**Comment by mikemcquaid** [comment]
Great work here :+1:

**Comment by sparrc** [comment]
@mikemcquaid Thanks for the tips, pushed the changes you suggested!

File: Library/Formula/telegraf.rb [hunk 3]
```diff
@@ -76,6 +76,8 @@ def plist; <<-EOS.undent
   end
 
   test do
-    assert_match /Telegraf/, shell_output("telegraf -version")
+    (testpath/"config.toml").write, shell_output("telegraf -sample-config")
```
Comment (mikemcquaid) [inline Library/Formula/telegraf.rb]: No `,` after `.write`

Comment (sparrc) [inline Library/Formula/telegraf.rb]: done

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**